agent/main.go

```diff
@@ -119,10 +119,10 @@ func newCmd() *cobra.Command {
 			}
 			s := settings{args[0], paths}
 			config, err := clientConfig.ClientConfig()
-			errors.CheckError(err)
+			errors.CheckErrorWithCode(err, errors.ErrorCommandSpecific)
```

Comment:
I think this can be simplified if line 115 uses `RunE` field instead of `Run`. Then this function can just return the error directly and the checking logic can be centralized in `main()`.

Reply:
`CheckErrorWithCode` just exits with the appropriate exit code (instead of the default exit code) -- there is no real error checking logic, we simply exit.

Reply:
Moving the logic of error handling to `main()` would assume that the `error` would actually hold non-generic information about the error, but I think currently most `error` variables are just generics, with nothing more than an error message produced by `fmt.Errorf()` or similar methods. But I agree, in the long run, we should have specialised `error` variables that enables to differentiate the errors directly from the `error` variable.

Reply:
I agree with the above. I just wanted to say that current approach might be less than ideal because:
- `defer`ed functions are not executed. Currently there is only one and it might not be a big deal. This might change in the future.
- Exiting on error might need something that is not accessible from a static method (e.g. a field or method on some object). E.g. the program may want to send an event to a metric/error aggregation system. Having multiple places that do this (vs one in the `main()`) might be more verbose.
- Having 1 entry point and 1 exit point is easier to follow in general, but this is subjective.